service/controller/v4/resource/chart/desired.go

```diff
@@ -74,6 +89,37 @@ func (r *Resource) getConfigMapValues(ctx context.Context, customObject v1alpha1
 	return chartValues, nil
 }
 
+func (r *Resource) getUserConfigMapValues(ctx context.Context, customObject v1alpha1.ChartConfig) (map[string]interface{}, error) {
+	userValues := make(map[string]interface{})
+
+	configMapName := key.UserConfigMapName(customObject)
+	configMapNamespace := key.UserConfigMapNamespace(customObject)
+
+	if configMapName != "" {
+		configMap, err := r.getConfigMap(ctx, configMapName, configMapNamespace)
+		if err != nil {
+			return userValues, microerror.Mask(err)
+		}
+
+		for k, v := range configMap.Data {
+			userValues[k] = v
+		}
+	}
+
+	return userValues, nil
```

Comment:
I would prefer this function to add root `configmap` key to the returned `userValues`, i.e. `map[string]interface{"configmap": userValues}`. Then names and merge functions would be more straight forward. Right now it's misleading IMO.

Reply:
Thanks, that's nice. I agree it will be clearer that way. I'll give it a try.

Reply:
Done.